src/sgame/botlib/bot_nav.cpp

```diff
@@ -427,7 +427,7 @@ void G_BotAddObstacle( const glm::vec3 &mins, const glm::vec3 &maxs, int obstacl
 		tempBox.mins[ 1 ] -= params->walkableHeight;
 
 		nav->cache->addBoxObstacle( tempBox.mins, tempBox.maxs, &ref );
-		obstacleHandles[i] = ref;
```

Comment:
Seems fishy that we are setting the same entry in the map on every loop iteration.

Reply:
That's true. In principle we could add obstacles per-class and it would still be nice to handle that. Coincidentally this seems to work so idk.

Reply:
So it works because if you have added the same number of obstacles to each mesh then you get the same handle? If obstacleHandles is already populated for the given key, this assumption should be checked with a warning or assertion.

Reply:
I just pushed something which uses arrays. So this should no longer be a problem.